concent_api/gatekeeper/views.py

```diff
@@ -1,11 +1,77 @@
 from base64                         import b64decode
 
+from django.conf                    import settings
 from django.http                    import HttpResponse
-from django.views.decorators.http   import require_POST
 from django.views.decorators.csrf   import csrf_exempt
+from django.views.decorators.http   import require_POST
+from django.views.decorators.http   import require_GET
+
+from gatekeeper.utils               import gatekeeper_access_denied_response
+
+from golem_messages.message         import init_messages
+from golem_messages.message         import Message
+from golem_messages.shortcuts       import load
 
 
 @csrf_exempt
 @require_POST
-def file_transfer_auth(request):
-    return HttpResponse("", status=200)
+def upload(request):
+    if request.content_type != 'application/x-www-form-urlencoded':
+        return gatekeeper_access_denied_response('Unsupported content type.')
+
+    init_messages()
+    response = parse_headers(request)
+    if response is not None:
+        return response
+
+    return HttpResponse("", status = 200)
+
+
+@csrf_exempt
+@require_GET
+def download(request):
+    if request.content_type == 'application/x-www-form-urlencoded':
+        return gatekeeper_access_denied_response('Unsupported content type.')
+
+    init_messages()
+    response = parse_headers(request)
+    if response is not None:
+        return response
+
+    return HttpResponse("", status = 200)
+
+
+def parse_headers(request):
+    # Decode and check if request header contains a golem message:
+    if 'HTTP_AUTHORIZATION' not in request.META:
+        return gatekeeper_access_denied_response("Missing 'Authorization' header.")
+
+    authorization_scheme_and_token = request.META['HTTP_AUTHORIZATION'].split(" ", 1)
+    assert len(authorization_scheme_and_token) in [1, 2]
+
+    if len(authorization_scheme_and_token) == 1:
+        return gatekeeper_access_denied_response("Missing token in the 'Authorization' header.")
+
+    (scheme, token) = authorization_scheme_and_token
+
+    if scheme != 'Golem':
+        return gatekeeper_access_denied_response("Unrecognized scheme in the 'Authorization' header.")
+
+    try:
+        decoded_auth_header_content = b64decode(token)
+    except ValueError:
```

Comment:
1) Pass `validate=False` to `b64decode()`

2) Why are you expecting `ValueError` here? The [docs for b64decode()](https://docs.python.org/3.6/library/base64.html#base64.b64decode) say that you will get `binascii.Error` if there are problems with padding or invalid characters in the input.

Reply:
Oops, I meant `validate=True` :) Sorry for the confusion.